website/content/posts/a-taste-of-policies.md

```diff
@@ -0,0 +1,232 @@
+Tekton Pipeline    v0.47.0
+Kubernetes Client  v0.27.4
+{{</code>}}
+
+The most basic verification we can do on this image is to verify its signature and signed SLSA
+Provenance were created by the expected identity:
+
+{{<code bash>}}
+ec validate image --policy '' --image $IMAGE \
+  --certificate-identity-regexp='https:\/\/github\.com\/(slsa-framework\/slsa-github-generator|lcarva\/festoji)\/' \
+  --certificate-oidc-issuer='https://token.actions.githubusercontent.com' \
+  --output yaml
+{{</code>}}
+
+A regular expression is needed for the certificate identity because the signature and the signed
+SLSA Provenance were created by different identities.
+
+## Adding a Policy
+
+The verification above is useful, but we want more. We want to apply policy rules to the signature
+and to the SLSA Provenance. We also want to more easily specify the values for the certificate
+flags. Let's create a policy!
+
+{{<code yaml>}}
+---
+identity:
+  subjectRegExp: >-
+    https:\/\/github\.com\/(slsa-framework\/slsa-github-generator|lcarva\/festoji)\/
+  issuer: https://token.actions.githubusercontent.com
+
+sources:
+  - policy:
+      - github.com/enterprise-contract/ec-policies//policy/lib
+      - github.com/enterprise-contract/ec-policies//policy/release
+    ruleData:
+      allowed_gh_workflow_repos:
+        - lcarva/festoji
+      allowed_gh_workflow_refs:
+        - refs/heads/master
+      allowed_gh_workflow_names:
+        - Package
+      allowed_gh_workflow_triggers:
+        - push
+
+configuration:
+  include:
+    - github_certificate
+{{</code>}}
+
+This policy moves the certificate flags to the policy itself. It also specifies certain policy rules
+to be executed. Here we are including some of the existing Enterprise Contract policy rules,
+[github_certificate](https://enterprisecontract.dev/docs/ec-policies/release_policy.html#github_certificate_package).
+These policy rules rely on certain data to be provided, e.g. the expected GitHub Workflow
+repository. With this policy saved as `policy.yaml`, we can simplify how the CLI is invoked:
+
+{{<code bash>}}
+ec validate image --policy 'policy.yaml' --image $IMAGE --output yaml
+{{</code>}}
+
+We can tweak the values of the `ruleData` in the policy to see what a failure would look like:
+
+{{<code yaml>}}
+  violations:
+  - metadata:
+      code: github_certificate.gh_workflow_name
+    msg: 'Name "Package" not in allowed list: ["spam"]'
+  - metadata:
+      code: github_certificate.gh_workflow_ref
+    msg: 'Ref "refs/heads/master" not in allowed list: ["refs/heads/spam"]'
+  - metadata:
+      code: github_certificate.gh_workflow_repository
+    msg: 'Repository "lcarva/festoji" not in allowed list: ["spam/spam"]'
+  - metadata:
+      code: github_certificate.gh_workflow_trigger
+    msg: 'Trigger "push" not in allowed list: ["spam"]'
+{{</code>}}
+
+As you can see, the Enterprise Contract provides a comprehensive list of all the identified
+violations.
```

Comment:
Idea to consider (if you didn't already). Add the `--info` flag for more detailed output, i.e. the rule titles and descriptions.

Reply:
Yeah, that's a good idea. Thanks!